src/main/java/org/broadinstitute/consent/http/resources/SwaggerResource.java

```diff
@@ -1,138 +1,144 @@
-                String content = FileUtils.readAllTextFromResource(swaggerResource + path);
-                if (content != null) {
-                    response = Response.ok().entity(content).type(mediaType).build();
-                } else {
-                    response = Response.status(Response.Status.NOT_FOUND).build();
-                }
-            }
+          response = Response.status(Response.Status.NOT_FOUND).build();
         }
-        return response;
-    }
-
-    private String getMediaTypeFromPath(String path) {
-        String mediaType;
-        switch (StringUtils.substringAfterLast(path, ".")) {
-            case "css":
-                mediaType = MEDIA_TYPE_CSS;
-                break;
-            case "js":
-                mediaType = MEDIA_TYPE_JS;
-                break;
-            case "png":
-                mediaType = MEDIA_TYPE_PNG;
-                break;
-            case "gif":
-                mediaType = MEDIA_TYPE_GIF;
-                break;
-            default:
-                mediaType = MediaType.TEXT_HTML;
-                break;
+      } else {
+        String content = FileUtils.readAllTextFromResource(swaggerResource + path);
+        if (content != null) {
+          response = Response.ok().entity(content).type(mediaType).build();
+        } else {
+          response = Response.status(Response.Status.NOT_FOUND).build();
         }
-        return mediaType;
+      }
     }
+    return response;
+  }
 
-    private String getIndex(String swaggerResource) {
-        String content = FileUtils.readAllTextFromResource(swaggerResource + "index.html");
-        return content
-            .replace("your-client-id", config.getClientId())
-            .replace("your-client-secret-if-required", "")
-            .replace("your-realms", "Broad Institute")
-            .replace("your-app-name", "Consent")
-            .replace("scopeSeparator: \",\"", "scopeSeparator: \" \"")
-            .replace("jsonEditor: false,", "jsonEditor: false," + "validatorUrl: null, apisSorter: \"alpha\", operationsSorter: \"alpha\",")
-            .replace("url = \"http://petstore.swagger.io/v2/swagger.json\";", "url = '/api-docs/api-docs.yaml';");
+  private String getMediaTypeFromPath(String path) {
+    String mediaType;
+    switch (StringUtils.substringAfterLast(path, ".")) {
+      case "css":
+        mediaType = MEDIA_TYPE_CSS;
+        break;
+      case "js":
+        mediaType = MEDIA_TYPE_JS;
+        break;
+      case "png":
+        mediaType = MEDIA_TYPE_PNG;
+        break;
+      case "gif":
+        mediaType = MEDIA_TYPE_GIF;
+        break;
+      default:
+        mediaType = MediaType.TEXT_HTML;
+        break;
     }
+    return mediaType;
+  }
 
+  private String getIndex(String swaggerResource) {
+    String content = FileUtils.readAllTextFromResource(swaggerResource + "index.html");
+    return content
+        .replace(
+            "url: \"https://petstore.swagger.io/v2/swagger.json\",",
```

Comment:
More of a question than a comment: Is there no way to configure Swagger to set these attributes?

Reply:
The swagger ui web jar has a pre-constructed html page with an existing swagger configuration. In the implementation we have, we pull that index.html file out and rewrite it with custom values. This has been the approach we've taken since we started using the web jar. I do think this is the simplest way to get our custom configurations into that html page. We've been doing it this way for quite a few years though, so there certainly could be a better way. Here's how we do this in our scala artifacts: https://github.com/broadinstitute/firecloud-orchestration/blob/develop/src/main/scala/org/broadinstitute/dsde/firecloud/FireCloudServiceActor.scala#L226